tests/balena-api-mock.ts

```diff
@@ -16,70 +16,122 @@
  */
 
 import * as _ from 'lodash';
-import * as nock from 'nock';
+import * as path from 'path';
 
-export class BalenaAPIMock {
-	public static basePathPattern = /api\.balena-cloud\.com/;
-	public readonly scope: nock.Scope;
-	// Expose `scope` as `expect` to allow for better semantics in tests
-	public readonly expect = this.scope;
+import { NockMock, ScopeOpts } from './nock-mock';
 
-	// For debugging tests
-	get unfulfilledCallCount(): number {
-		return this.scope.pendingMocks().length;
-	}
+const apiResponsePath = path.normalize(
+	path.join(__dirname, 'test-data', 'api-response'),
+);
+
+const jHeader = { 'Content-Type': 'application/json' };
 
+export class BalenaAPIMock extends NockMock {
 	constructor() {
-		nock.cleanAll();
+		super('https://api.balena-cloud.com');
+	}
+
+	public expectGetApplication(opts: ScopeOpts = {}) {
```

Comment:
Good idea re standardisation.   I think we might start having some issues once we continue to add more of these methods though to this file though.  E.g. when I first added `expectWhoAmI` and `expectMixpanel`, I put them in `balena-api-mock.ts` for convenience, because they are used in every/most CLI actions.  I think this file will get huge and hard to manage if we keep putting them here though (and will obscure the general/framework parts).  For the ones that are only used in one or a few actions, maybe we should brainstorm about where would be the most appropriate place for these to live.  Maybe in or nearer the particular test specs themselves? E.g. perhaps something like a `call-helpers.ts` file in each test spec folder?